spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +71,16 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	@PostConstruct
+	public void deduceModuleArguments() throws Exception {
```

Comment:
If we know the specific order of the modules, we can handle the basic single-flow Source/Processor(s)/Sink easily - we can do that as a separate discussion though :). Let's just assume that we want to preserve ordering :)

Summary of a discussion with @ericbottard: it may be possible to rely on `[]` or `.` in environment variables for CF, as the format restrictions apply to shell only (see http://stackoverflow.com/questions/2821043/allowed-characters-in-linux-environment-variable-names).